src/org/labkey/test/components/ui/entities/EntityInsertPanel.java

```diff
@@ -50,6 +50,21 @@ public WebDriver getDriver()
         return _driver;
     }
 
+    @Override
+    protected void waitForReady()
+    {
+        WebDriverWrapper.waitFor(()-> {
+            try{
+                return isFileUploadVisible() ||
+                        isGridVisible() ||
+                        hasTabs();
```

Comment:
I don't think we want the `hasTabs` check here. That could make this check succeed before the grid or file upload are actually ready.

Reply:
OK.
Also it occurs to me that as we add waitForReady to query-backed components, it makes sense to make the ready timeout settable